extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeMixPlaylistExtractor.java

```diff
@@ -0,0 +1,228 @@
+import javax.annotation.Nullable;
+
+import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.extractCookieValue;
+import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.getJsonResponse;
+import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.getResponse;
+import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.getUrlFromNavigationEndpoint;
+import static org.schabi.newpipe.extractor.services.youtube.YoutubeParsingHelper.toJsonArray;
+
+/**
+ * A {@link YoutubePlaylistExtractor} for a mix (auto-generated playlist). It handles URLs in the format of
+ * <code>youtube.com/watch?v=videoId&list=playlistId"</code>
+ */
+public class YoutubeMixPlaylistExtractor extends PlaylistExtractor {
+
+    /**
+     * YouTube identifies mixes based on this cookie. With this information it can generate
+     * continuations without duplicates.
+     */
+    private static final String COOKIE_NAME = "VISITOR_INFO1_LIVE";
+
+    private JsonObject initialData;
+    private JsonObject playlistData;
+    private String cookieValue;
+
+    public YoutubeMixPlaylistExtractor(final StreamingService service,
+                                       final ListLinkHandler linkHandler) {
+        super(service, linkHandler);
+    }
+
+    @Override
+    public void onFetchPage(@Nonnull final Downloader downloader)
+            throws IOException, ExtractionException {
+        final String url = getUrl() + "&pbj=1";
+        final Response response = getResponse(url, getExtractorLocalization());
+        final JsonArray ajaxJson = toJsonArray(response.responseBody());
+        initialData = ajaxJson.getObject(3).getObject("response");
+        playlistData = initialData.getObject("contents").getObject("twoColumnWatchNextResults")
+                .getObject("playlist").getObject("playlist");
+        cookieValue = extractCookieValue(COOKIE_NAME, response);
+    }
+
+    @Nonnull
+    @Override
+    public String getName() throws ParsingException {
+        final String name = playlistData.getString("title");
+        if (name == null) {
+            throw new ParsingException("Could not get playlist name");
+        }
+        return name;
+    }
+
+    @Override
+    public String getThumbnailUrl() throws ParsingException {
+        try {
+            final String playlistId = playlistData.getString("playlistId");
+            try {
+                return getThumbnailUrlFromPlaylistId(playlistId);
+            } catch (ParsingException e) {
+                //fallback to thumbnail of current video. Always the case for channel mix
+                return getThumbnailUrlFromVideoId(
+                    initialData.getObject("currentVideoEndpoint").getObject("watchEndpoint")
+                        .getString("videoId"));
+            }
+        } catch (Exception e) {
+            throw new ParsingException("Could not get playlist thumbnail", e);
+        }
```

Comment:
Do this, so that the original error is reported if even the fallback fails
```suggestion
        try {
            return getThumbnailUrlFromPlaylistId(playlistData.getString("playlistId"));
        } catch (final Exception e) {
            try {
                //fallback to thumbnail of current video. Always the case for channel mix
                return getThumbnailUrlFromVideoId(
                    initialData.getObject("currentVideoEndpoint").getObject("watchEndpoint")
                        .getString("videoId"));
            } catch (final Exception ignored) {
            }
            throw new ParsingException("Could not get playlist thumbnail", e);
        }
```